metadataproxy/roles.py

```diff
@@ -157,117 +157,129 @@ def find_container(ip):
         'InstanceProfileArn': role['AssumedRoleUser']['Arn'],
         'InstanceProfileId': role['AssumedRoleUser']['AssumedRoleId']
     }
 
 
-def get_role_arn(role_name):
-    # Role name is an arn. Just return it.
-    if role_name.startswith('arn:aws'):
-        return role_name
-    # Role name includes an account name/id, split them
-    if '@' in role_name:
-        assume_role, account_name = role_name.split('@')
-    # No role name/id, try to get the default account id
+def get_role_arn(role_params):
+    if role_params['account_id']:
+        # Try to map the name to an account ID. If it isn't found, assume an ID was passed
+        # in and use it as-is.
+        role_params['account_id'] = app.config['AWS_ACCOUNT_MAP'].get(
+            role_params['account_id'],
+            role_params['account_id']
+        )
     else:
-        assume_role = role_name
         if app.config['DEFAULT_ACCOUNT_ID']:
-            account_name = app.config['DEFAULT_ACCOUNT_ID']
+            role_params['account_id'] = app.config['DEFAULT_ACCOUNT_ID']
         # No default account id defined. Get the ARN by looking up the role
         # name. This is a backwards compat use-case for when we didn't require
         # the default account id.
         else:
             iam = iam_client()
             try:
                 with PrintingBlockTimer('iam.get_role'):
-                    role = iam.get_role(RoleName=role_name)
+                    role = iam.get_role(RoleName=role_params['name'])
                     return role['Role']['Arn']
             except ClientError as e:
                 response = e.response['ResponseMetadata']
                 raise GetRoleError((response['HTTPStatusCode'], e.message))
-    # Map the name to an account ID. If it isn't found, assume an ID was passed
-    # in and use that.
-    account_id = app.config['AWS_ACCOUNT_MAP'].get(account_name, account_name)
     # Return a generated ARN
-    return 'arn:aws:iam::{0}:role/{1}'.format(account_id, assume_role)
+    return 'arn:aws:iam::{account_id}:role/{name}'.format(**role_params)
 
 
 @log_exec_time
-def get_assumed_role(requested_role):
-    if requested_role in ROLES:
-        assumed_role = ROLES[requested_role]
+def get_assumed_role(role_params):
+    arn = get_role_arn(role_params)
+    if arn in ROLES:
+        assumed_role = ROLES[arn]
         expiration = assumed_role['Credentials']['Expiration']
         now = datetime.datetime.now(dateutil.tz.tzutc())
         expire_check = now + datetime.timedelta(minutes=5)
         if expire_check < expiration:
             return assumed_role
-    arn = get_role_arn(requested_role)
     with PrintingBlockTimer('sts.assume_role'):
         sts = sts_client()
-        assumed_role = sts.assume_role(
-            RoleArn=arn,
-            RoleSessionName='devproxyauth'
-        )
-    ROLES[requested_role] = assumed_role
+        kwargs = {'RoleArn': arn, 'RoleSessionName': (role_params['session_name'] or 'devproxyauth')}
```

Comment:
Can you break this or conditional out into a variable assignment above this, then assign RoleSessionName the value of the variable? I'm ok with the extra verbose code for better readability.